homework/index.js

```diff
@@ -28,14 +28,53 @@
     });
     return elem;
   }
-
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
-  }
+    const li = createAndAppend('li', ul, { class: 'item' });
+    const table = createAndAppend('table', li);
+
+    // add Repository name
+    const row1 = createAndAppend('tr', table);
+    const repoName = createAndAppend('th', row1);
+    repoName.innerText = 'Repository:';
+    const repoNameContent = createAndAppend('td', row1);
+    createAndAppend('a', repoNameContent, {
+      href: repo.html_url,
+      text: repo.name,
+    });
+
+    // add description
+    const row2 = createAndAppend('tr', table);
+    const repoDescription = createAndAppend('th', row2);
+    repoDescription.innerText = 'Description:';
+    const repoDescriptionContent = createAndAppend('td', row2);
+    if (repo.description) {
+      repoDescriptionContent.innerText = repo.description;
+    } else {
+      repoDescriptionContent.innerText = 'N/A';
+    }
 
+    // add Forks
+    const row3 = createAndAppend('tr', table);
+    const repoForks = createAndAppend('th', row3);
+    repoForks.innerText = 'Forks:';
+    const repoForksContent = createAndAppend('td', row3);
+    repoForksContent.innerText = repo.forks_count;
+
+    // add updated
+    const row4 = createAndAppend('tr', table);
+    const repoUpdate = createAndAppend('th', row4);
+    repoUpdate.innerText = 'Updated:';
+    const repoUpdateContent = createAndAppend('td', row4);
+    repoUpdateContent.innerText = new Date(repo.updated_at).toLocaleString();
+  }
```

Comment:
I see some repetition here: create a `tr`, add a `th` child with some label text and a `td` child with a repository property value. Can you write a function for this and call it for each row to be created?

The function could look like this:

```js
function addRow(table, labelText, value) {
  const tr = ...;
  createAndAppend('th', table, ....);
  createAndAppend('td, table, ... );
  return tr;
}
```

You could use this function to create the table rows, instead of repeating the code four times.

For the first (that is to contain an `<a>` tag instead of simple text) you would need to do something special after calling the `addRow()` function:

```js
const firstRow = addRow(table, 'Repository:', '');
createAndAppend('a', firstRow.lastChild, {
  href: repo.html_url,
  text: repo.name,
  target: "_blank",
});
```

Note that the third parameter for `addRow` is an empty string in this case, because we don't actually want to set the text content.

Reply:
Thanks alot for this function, it's very hulpful :)